test/warnings/make.jl

```diff
@@ -0,0 +1,62 @@
+module WarningTests
+
+# Test warnings
+# see https://github.com/JuliaDocs/Documenter.jl/issues/2805
+
+using Test
+using Documenter
+using IOCapture
+
+isdefined(Main, :Documenter) || @eval Main import Documenter
+
+@testset "at-doc warnings" begin
+
+    captured = IOCapture.capture() do
+        makedocs(;
+            sitename = "at-docs",
+            pages = ["at-docs.md"],
+            warnonly = true,
+            format = Documenter.HTML(
+                prettyurls = false,
+                inventory_version = "",
+            ),
+        )
+    end
+    @test isnothing(captured.value)
+
+    # sanitize the output
+    output = captured.output
+    output = replace(output, r"\@ Documenter.*" => "@ Documenter")
+    output = replace(output, "src\\" => "src/")
+    @test output ==
+        """
+        [ Info: SetupBuildDirectory: setting up build directory.
+        [ Info: Doctest: running doctests.
+        [ Info: ExpandTemplates: expanding markdown templates.
+        ┌ Warning: undefined binding 'Base.nonsenseBindingThatDoesNotExist' in `@docs` block in src/at-docs.md:4-6
+        │ ```@docs
+        │ Base.nonsenseBindingThatDoesNotExist()
+        │ ```
+        └ @ Documenter
+        ┌ Warning: no docs found for 'Base.sin()' in `@docs` block in src/at-docs.md:9-11
+        │ ```@docs
+        │ Base.sin()
+        │ ```
+        └ @ Documenter
+        ┌ Warning: failed to evaluate `Base.sin(::NonsenseTypeThatDoesNotExist)` in `@docs` block in src/at-docs.md:14-16
+        │ ```@docs
+        │ Base.sin(::NonsenseTypeThatDoesNotExist)
+        │ ```
+        │   exception =
+        │    UndefVarError: `NonsenseTypeThatDoesNotExist` not defined in `Main`
+        │    Suggestion: check for spelling errors or missing imports.
+        └ @ Documenter
+        [ Info: CrossReferences: building cross-references.
+        [ Info: CheckDocument: running document checks.
+        [ Info: Populate: populating indices.
+        [ Info: RenderDocument: rendering document.
+        [ Info: HTMLWriter: rendering HTML pages.
+        """
```

Comment:
This test and the output here is more or less what I have in mind for addressing issue #2805. I'd imagine adding a bunch of additional `.md` files to deal with warnings in `@example`, `@repl`, etc. blocks, and beyond. 

The check here is perhaps too fragile, the Info messages before and after should perhaps not be part of the text as they might change over time (?) but for now it was convenient to have them.


I am actually wondering if it wouldn't be better to have this all be in a `jldoctest`, so that one can use `doctest=:fix` to update it for changes in the output (e.g. tweaked line numbers)